Darkmira/drop-master
# Master RestApi, websocket push events, send orders to robots

## Install this PR

Requires git, Docker, docker-compose.

``` bash
git clone git@github.com:Darkmira/drop-master.git --branch feature/init
cd drop-master/

# Create your .env from distribution and edit it
cp .env.dist .env

make
```

> **Note**: Sometimes you'll need to do either a
> `chown -R {your_user}:{your_group} .`
> or a
> `chmod -R 777 var/*`
> to make it work.


## Usage

Run all with Docker with command `make`.

Docker runs the whole environment, the RestApi, the websocket server and PHPMyAdmin. You now have access to:

 - http://0.0.0.0:8480/index-dev.php/api/race Rest Api (dev mode)
 - http://0.0.0.0:8480/api/race Rest Api (prod mode)
 - http://0.0.0.0:8480/index-dev.php/_profiler/ Symfony web profiler (only dev mode).
 - http://0.0.0.0:8481 PHPMyAdmin (login: `root` / `root`).
 - `ws://0.0.0.0:8482` Websocket server.


### Postman collection

You can use Postman to send predefined requests to the api.

Import the postman collection which is in this PR, the file `postman.json` at the root.

## Test this PR

- Use predefined request in Postman collection to:
    - POST orders
    - GET race state
- Test websocket push events:
    - Go to http://0.0.0.0:8480/test-ws/index.html
    - Open Javascript console
    - POST new orders.
    - You should see websocket push event with the new race state.
- Run command to process vote every 20 seconds and send to fleet control api:
    - `make bash`
    - `bin/console drop:votes:schedule 20` (by default send to httpbin.org)
    - It should reset votes, and push the new race state to websocket clients (in JS console http://0.0.0.0:8480/test-ws/index.html)

### Simulate twitter observer

Use [this gist](https://gist.github.com/alcalyn/6963659e09dbbae82b6e7737fb0e76eb) to simulate twitter observer, and observe real tweets from twitter:

``` bash
git clone https://gist.github.com/alcalyn/6963659e09dbbae82b6e7737fb0e76eb twitter-observer
cd twitter-observer/

composer install

# Update .env to add your twitter keys

php run.php
```

Then tweet with the hashtag `#droprobot` something like:

`Hello ! team:green and also move:left #droprobot`

You should see yourself in http://0.0.0.0:8480/index-dev.php/api/race
